compiler+runtime/src/cpp/clojure/core_native.cpp

```diff
@@ -276,6 +276,27 @@ namespace clojure::core_native
   {
     return __rt_ctx->eval(expr);
   }
+
+// TODO: implement opts
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunused-parameter"
+  static object_ptr read_string(object_ptr const opts, object_ptr const str)
+#pragma clang diagnostic pop
+  {
+    return __rt_ctx->read_string(runtime::to_string(str));
+  }
+
+  static object_ptr hash_unordered_coll(object_ptr const o) {
+    return make_box(hash::unordered(o));
+  }
```

Comment:
For this an `compare` below, we shouldn't need a fn here. The `intern_fn` should be able to wrap `hash::unordered` and `runtime::compare` for us. Give it a go.

Reply:
You are correct, although `hash::unordered` took a bit of casting:

```cpp
static_cast<native_hash (*)(object const *)>(&hash::unordered)
```